scripts/vm/hypervisor/kvm/nsrkvmbackup.sh

```diff
@@ -0,0 +1,270 @@
+
+
+OPTIND=1
+
+verb=0
+logDir="/nsr/logs/cloudstack/"
+snapPrefix="CSBKP_$RANDOM_"
+clusterClient=""
+networkerServer=""
+hvVersion=""
+libvVersion=""
+apiVersion=""
+kvmDName=""
+kvmDUuid=""
+logFile=""
+mediaPool=""
+retentionTime=""
+
+
+log () {
+
+    [[ "$verb" -eq 1 ]] && builtin echo "$@"
+    if [[ "$1" == "-ne"  || "$1" == "-e" || "$1" == "-n" ]]; then
+          builtin echo -e "$(date '+%Y-%m-%d %H-%M-%S>')" "${@: 2}" >> "$logFile"
+    else
+          builtin echo "$(date '+%Y-%m-%d %H-%M-%S>')" "$@" >> "$logFile"
+    fi
+
+}
+
+vercomp(){
+   local a b IFS=. -; set -f
+   printf -v a %08d $1; printf -v b %08d $3
+   test $a "$2" $b
+}
+usage() {
+        echo "
+
+Usage:[-v] [-h] [-l log_dir] [-dr] [-s networker_server] [-c networker_cluster_client] [-t target_vm] [-u target_uuid] [-p snapprefix] [-P media_pool ] [-R retention_time ]
+
+Options:
+        -h Help and usage
+        -v Enable verbose mode
+        -l log_dir. Specify log directory. Default is /nsr/logs/cloudstack
+        -s networker_server  Specifiy the EMC Networker server we are going to use
+        -c networker_cluster_client  Specify the EMC Networker client CLUSTER to use
+        -t target_vm KVM domain to backup
+        -u target_uuid KVM domain to backup
+        -p Snapshot Prefix for backups
+        -P mediaPool EMC Networker Media Pool
+        -R retention_time Backup retention time
+
+Supplements Apache Cloudstack B&R Framework  EMC Networker plugin and performs the backup of the Virtual Machines
+"
+
+}
+
+sanity_checks() {
+
+        log "Performing environment sanity checks..."
+        log -ne "\t[1] Checking if Networker is installed\t"
+        if [[ $(systemctl list-unit-files | grep networker) = *networker* ]]; then
+                        log "Success"
+        else
+                        log "Failure"
+                        log -e "\n\tNetworker Service NOT FOUND. Make sure that Networker client is properly installed"
+                        exit 1
+        fi
+
+        log -ne "\t[2] Checking if Networker is running\t"
+        if [[ $(systemctl is-active networker) = *active* ]]; then
+                        log "Success"
+        else
+                        log "Failure"
+                        log -e "\n\tNetworker Service is not running. Investigate Networker logs, startup server and try again"
+                        exit 2
+        fi
+        log -ne "\t[3] Checking Networker DNS Resolution\t"
+        if [[ $(getent hosts "$networkerServer") = *$networkerServer* ]]; then
+                        log "Success"
+        else
+                        log "Failure"
+                        log -e "\n\tNetworker Server cannot be resolved. Backups will most probably fail. Consider adding the ip/hostname to /etc/host or fix DNS resolution"
+                        exit 3
+        fi
+
+        log -ne "\t[4] Checking QEMU / Libvirt Versions \t"
+        hvVersion=$(virsh version | grep hypervisor | awk '{print $(NF)}')
```

Comment:
I would avoid mixing information one can execute via java/libvirt-library; is this run on KVM agents on or mgmt server?

Reply:
It is running on KVM Agents.